src/parser/core/modules/Abilities.ts

```diff
@@ -135,6 +152,61 @@ class Abilities extends Module {
     return ability ? ability.charges || 1 : undefined;
   }
 
+  protected updateMaxCharges(spellId: number, maxCharges: number) {
+    const abilityIndex = this.getAbilityIndex(spellId);
+
+    if (abilityIndex === -1) {
+      return;
+    }
+
+    // The amount of charges should never drop below 1.
+    if (maxCharges < 1) {
+      maxCharges = 1;
+    }
+
+    this.activeAbilities[abilityIndex].charges = maxCharges;
+  }
+
+  increaseMaxCharges(event: AnyEvent, spellId: number, increaseBy: number = 1) {
+    const currentCharges = this.getMaxCharges(spellId);
+
+    if (currentCharges === undefined) {
+      return;
+    }
+
+    this.updateMaxCharges(spellId, currentCharges + increaseBy);
+
+    this.eventEmitter.fabricateEvent(
+      {
+        type: EventType.MaxChargesIncreased,
+        timestamp: event.timestamp,
+        spellId: spellId,
+        by: increaseBy,
+      },
+      event,
+    );
+  }
+
+  decreaseMaxCharges(event: AnyEvent, spellId: number, decreaseBy: number = 1) {
+    const currentCharges = this.getMaxCharges(spellId);
+
+    if (currentCharges === undefined) {
+      return;
+    }
+
+    this.updateMaxCharges(spellId, currentCharges - decreaseBy);
+
+    this.eventEmitter.fabricateEvent(
+      {
+        type: EventType.MaxChargesDecreased,
+        timestamp: event.timestamp,
+        spellId: spellId,
+        by: decreaseBy,
+      },
+      event,
+    );
+  }
```

Comment:
I'd prefer `increaseBy` and `decreaseBy` were required fields, just because I saw them being called in Dark Thoughts and immediately thought "INCREASED BY HOW MUCH?". IMO, defaulting to 1 hurts readability.

Reply:
I made them mandatory as you suggested.